### PR TITLE
[MODEXPS-154]. Export does not run when "Weekly" frequency was specified in scheduling

### PR DESCRIPTION
**Pull request by SerhiiNosko**
## Purpose
https://issues.folio.org/browse/MODEXPS-154

### CODE REVIEW

File: src/main/java/org/folio/des/scheduling/acquisition/AcqBaseExportTaskTrigger.java [hunk 1]
```diff
@@ -85,43 +87,48 @@ private Date getNextTime(Date lastActualExecutionTime) {
...
-          break;
+  private ZonedDateTime findNextDayOfWeek(ZonedDateTime scheduleDateTime) {
+    List<DayOfWeek> sortedChosenDays = normalizeAndSortDayOfWeek();
+
+    ZonedDateTime currentDateTime = getNowDateTime(scheduleParameters.getTimeZone());
+    DayOfWeek currentDay = currentDateTime.getDayOfWeek();
+
+    if (CollectionUtils.isNotEmpty(sortedChosenDays)) {
+      // iterate thought all days from Mon to Sun
+      for (DayOfWeek day: DayOfWeek.values()) {
```
Comment (NikitaSedyx) [inline src/main/java/org/folio/des/scheduling/acquisition/AcqBaseExportTaskTrigger.java:106]: why do we need this loop?

File: src/main/java/org/folio/des/scheduling/acquisition/AcqBaseExportTaskTrigger.java [hunk 2]
```diff
@@ -85,43 +87,48 @@ private Date getNextTime(Date lastActualExecutionTime) {
...
+        // iterate thought chosen days
+        for (DayOfWeek chosenDay : sortedChosenDays) {
+          if (day == chosenDay) {
+            if (currentDay == chosenDay) {
+              if (currentDateTime.isBefore(scheduleDateTime)) {
+                // run scheduler in the same day
+                return scheduleDateTime;
+              } else {
+                return scheduleDateTime.plusDays(7); // schedule after a week
```
Comment (NikitaSedyx) [inline src/main/java/org/folio/des/scheduling/acquisition/AcqBaseExportTaskTrigger.java:115]: possibly it's another issue: user selected Friday and Saturday 10am at 11am, job should be started on Saturday 10am, but will be started next Friday 10am

Comment (SerhiiNosko) [reply src/main/java/org/folio/des/scheduling/acquisition/AcqBaseExportTaskTrigger.java:115]: thanks, writing unit test to cover this test case and possibly adjust business logic

File: src/main/java/org/folio/des/scheduling/acquisition/AcqBaseExportTaskTrigger.java (hunk 1 above)
Comment (SerhiiNosko) [reply src/main/java/org/folio/des/scheduling/acquisition/AcqBaseExportTaskTrigger.java:106]: It actually can be deleted, forgot to delete from initial implementation, thanks

**Review by okolawole-ebsco — APPROVED** [review]

File: src/main/java/org/folio/des/scheduling/acquisition/AcqBaseExportTaskTrigger.java [hunk 3]
```diff
@@ -85,43 +87,43 @@ private Date getNextTime(Date lastActualExecutionTime) {
...
-          nextDayOfWeek = dayOfWeek;
-          break;
+  private ZonedDateTime findNextDayOfWeek(ZonedDateTime scheduleDateTime) {
+    List<DayOfWeek> sortedChosenDays = normalizeAndSortDayOfWeek();
+
+    ZonedDateTime currentDateTime = getNowDateTime(scheduleParameters.getTimeZone());
+    DayOfWeek currentDay = currentDateTime.getDayOfWeek();
+
+    if (CollectionUtils.isNotEmpty(sortedChosenDays)) {
+      // iterate thought chosen days
```
[the remark this answers is not in the document]
Comment (SerhiiNosko) [reply src/main/java/org/folio/des/scheduling/acquisition/AcqBaseExportTaskTrigger.java]: changed, thanks

**Review by Abdulkhakimov — APPROVED** [review]

**Review by khandramai — APPROVED** [review]

**Pull request merged**